src/coap_client_libcoap.c

```diff
@@ -531,37 +559,20 @@ static void add_observation(golioth_coap_request_msg_t *req, struct golioth_clie
     if (!found_slot)
     {
         GLTH_LOGE(TAG, "Unable to observe path %s, no slots available", req->path);
-        return;
+        return GOLIOTH_ERR_QUEUE_FULL;
     }
 
-    obs_info->in_use = true;
-    memcpy(&obs_info->req, req, sizeof(obs_info->req));
-}
-
-static void golioth_coap_observe(golioth_coap_request_msg_t *req,
-                                 struct golioth_client *client,
-                                 coap_session_t *session)
-{
-    // GET with an OBSERVE option
-    coap_pdu_t *req_pdu = coap_new_pdu(COAP_MESSAGE_CON, COAP_REQUEST_CODE_GET, session);
-    if (!req_pdu)
+    int err = golioth_coap_observe(req, client, session);
+    if (err == COAP_INVALID_MID)
```

Comment:
What about other errors?

Reply:
This is the only error code that `send_coap()` returns ([libcoap docs](https://libcoap.net/doc/reference/4.3.4/man_coap_send.html#coap_send)). This issue here isn't the error testing, but where I'm doing it.

I've moved that check up into `golioth_coap_observe()`, left a comment about the error code, and changed the return type of the functions to be `enum golioth_status`.